tests/templates/test_subroutines.py

```diff
@@ -800,3 +785,146 @@ def circuit(w_ph_0, w_ph_1, w_pphh):
+        """Tests if the correct errors are thrown when the user attempts to input a matrix with non-Pauli terms"""
+
+        n_wires = 2
+        dev = qml.device('default.qubit', wires=n_wires)
+
+        @qml.qnode(dev)
+        def circuit():
+            ApproxTimeEvolution(hamiltonian, 2, 3)
+            return [qml.expval(qml.PauliZ(wires=i)) for i in range(n_wires)]
+
+        with pytest.raises(ValueError, match="hamiltonian must be written in terms of Pauli matrices, got '{}'".format(output)):
+            circuit()
+
+    @pytest.mark.parametrize(
+        ("time", "hamiltonian", "steps", "gates"),
+        [
+            (2, qml.Hamiltonian([1, 1], [qml.PauliX(0), qml.PauliX(1)]), 2,
+             [
+                 qml.PauliRot(2.0, 'X', wires=[0]),
+                 qml.PauliRot(2.0, 'X', wires=[1]),
+                 qml.PauliRot(2.0, 'X', wires=[0]),
+                 qml.PauliRot(2.0, 'X', wires=[1])
+             ]
+             ),
+            (2, qml.Hamiltonian([2, 0.5], [qml.PauliX("a"), qml.PauliZ("b") @ qml.PauliX("a")] ), 2,
+             [
+                 qml.PauliRot(4.0, 'X', wires=["a"]),
+                 qml.PauliRot(1.0, 'ZX', wires=["b", "a"]),
+                 qml.PauliRot(4.0, 'X', wires=["a"]),
+                 qml.PauliRot(1.0, 'ZX', wires=["b", "a"])
+             ]
+             ),
+            (2, qml.Hamiltonian([1, 1], [qml.PauliX(0), qml.Identity(0) @ qml.Identity(1)]), 2,
+             [
+                 qml.PauliRot(2.0, 'X', wires=[0]),
+                 qml.PauliRot(2.0, 'X', wires=[0])
+             ]
+             ),
+            (2, qml.Hamiltonian([2, 0.5, 0.5], [qml.PauliX("a"), qml.PauliZ(-15) @ qml.PauliX("a"), qml.Identity(0) @ qml.PauliY(-15)]), 1,
+             [
+                 qml.PauliRot(8.0, 'X', wires=["a"]),
+                 qml.PauliRot(2.0, 'ZX', wires=[-15, "a"]),
+                 qml.PauliRot(2.0, "IY", wires=[0, -15])
+             ]
+             ),
+        ]
+    )
+    def test_evolution_operations(self, time, hamiltonian, steps, gates):
+        """Tests that the sequence of gates implemented in the ApproxTimeEvolution template is correct"""
+
+        n_wires = 2
+
+        with qml._queuing.OperationRecorder() as rec:
+            ApproxTimeEvolution(hamiltonian, time, steps)
+
+        for i, gate in enumerate(rec.operations):
+            prep = [gate.parameters, gate.wires]
+            target = [gates[i].parameters, gates[i].wires]
+
+            assert prep == target
+
+    @pytest.mark.parametrize(
+        ("time", "hamiltonian", "steps", "expectation"),
+        [
+            (np.pi, qml.Hamiltonian([1, 1], [qml.PauliX(0), qml.PauliX(1)]), 2, [1.0, 1.0]),
+            (np.pi/2, qml.Hamiltonian([0.5, 1], [qml.PauliY(0), qml.Identity(0) @ qml.PauliX(1)]), 1, [0.0, -1.0]),
+            (np.pi/4, qml.Hamiltonian([1, 1, 1], [qml.PauliX(0), qml.PauliZ(0) @ qml.PauliZ(1), qml.PauliX(1)]), 1, [0.0, 0.0])
+        ]
```

Comment:
Just as a final sanity check, I'd recommend having 1-2 test cases that are not nice multiples of pi :pie: